bin/stashcp2/stashcp.py

```diff
@@ -0,0 +1,219 @@
+        payload['timestamp']=end1
+        payload['host']=cache
+        payload['filename']=sourceFile
+        payload['filesize']=fileSize
+        payload['download_size']=dlSz
+        payload['download_time']=dltime
+        payload['sitename']=sitename
+        payload['destination_space']=destSpace
+        payload['status']=status
+        payload['xrdexit1']=xrd_exit
+        payload['xrdexit2']=xrdexit2
+        payload['xrdexit3']=xrdexit3
+        payload['tries']=tries
+        payload['xrdcp_version']=xrdcp_version
+        payload['start1']=start1
+        payload['end1']=end1
+        payload['start2']=start2
+        payload['end2']=end2
+        payload['start3']=start3
+        payload['cache']=cache
+        try:
+            p = multiprocessing.Process(target=es_send, name="es_send", args=(payload,))
+            p.start()
+            time.sleep(5)
+            p.terminate()
+        except:
+            print "Error curling to ES"
+    else: #Copy again using same cache
+        print "1st try failed on %s, trying again" % cache
+        date=datetime.datetime.now()
+        start2=int(time.mktime(date.timetuple()))*1000
+        copy=subprocess.Popen([command],stdout=subprocess.PIPE,shell=True)
+        xrd_exit=copy.communicate()[0].split()[-1]
+        date=datetime.datetime.now()
+        end2=int(time.mktime(date.timetuple()))*1000
+        dlSz=os.stat(filename).st_size
+        if xrd_exit=='0': #Worked second try
+            status = 'Success'
+            tries=2
+            dltime=end2-start2
+            payload={}
+            payload['timestamp']=end1
+            payload['host']=cache
+            payload['filename']=sourceFile
+            payload['filesize']=fileSize
+            payload['download_size']=dlSz
+            payload['download_time']=dltime
+            payload['sitename']=sitename
+            payload['destination_space']=destSpace
+            payload['status']=status
+            payload['xrdexit1']=xrd_exit
+            payload['xrdexit2']=xrdexit2
+            payload['xrdexit3']=xrdexit3
+            payload['tries']=tries
+            payload['xrdcp_version']=xrdcp_version
+            payload['start1']=start1
+            payload['end1']=end1
+            payload['start2']=start2
+            payload['end2']=end2
+            payload['start3']=start3
+            payload['cache']=cache
+            try:
+                p = multiprocessing.Process(target=es_send, name="es_send", args=(payload,))
+                p.start()
+                time.sleep(5)
+                p.terminate()
+            except:
+                print "Error curling to ES"
+        else: #Pull from origin
+            print "2nd try failed on %s, pulling from origin" % cache
+            cache="root://stash.osgconnect.net"
+            command = "python ./timeout.py -t "+str(TIMEOUT)+ " -f "+sourceFile + " -d "+str(DIFF)+" -s "+str(fileSize)+" -x "+str(xrdargs)+" -c "+cache+" -z "+destination
```

Comment:
Why are you calling another python script as a process.  Can't you just create a module, class, or function that does the time and takes all the necessary arguments?